app/code/Magento/ProductAlert/Controller/Unsubscribe/Price.php

```diff
@@ -51,8 +56,13 @@ public function execute()
             /* @var $product \Magento\Catalog\Model\Product */
             $product = $this->productRepository->getById($productId);
             if (!$product->isVisibleInCatalog()) {
-                throw new NoSuchEntityException();
+                $this->messageManager->addErrorMessage(
```

Comment:
It looks like separate feature. Would be great to cover it with tests.

Reply:
> It looks like separate feature. Would be great to cover it with tests.

We do not have unit tests for any of `Magento_ProductAlert` controllers.

Probably now you will ask me to create Unit Tests for each ones....

Reply:
Ok, let's leave it as is for now